satpy/enhancements/__init__.py

```diff
@@ -189,7 +189,8 @@ def _merge_colormaps(kwargs):
     else:
         for itm in palette:
             cmap = create_colormap(itm)
-            cmap.set_range(itm["min_value"], itm["max_value"])
+            if 'min_value' in itm:
+                cmap.set_range(itm["min_value"], itm["max_value"])
```

Comment:
Just to be clear, this means that if `min_value` is not defined that the colormap can assume data is between 0 and 1, right?

Reply:
Not really.  In the old code, if min_value and max_value were not given in the enhancements/generic.yaml file, the code would dump with a key_error for the missing min_value,max_value .. as opposing to assuming 0,1.  So I just kept that rather than defaulting them to 0,1 just to maintain what was there. Happy to change, though.

I'm also not too happy (now) with my adding the new kwarg of ctab_mode and only explicitly referencing "VRGB" to denote (value,r,g,b) with anything else defaulting to (r,g,b,a) as opposed to some explicit value (like RGBA) for that, and them some warning if a unrecognized value was given.  I'll  happily take comments for improvement on that too.

Reply:
For the 0,1 thing: I think that's what I meant if we're understanding each other. Basically with out the `set_range` call, the values from the colormap are used which are supposed to be 0 to 1. I think.

I think I'd prefer `colormap_mode` or just `mode` (if it is in a colormap-specific section of the config) for the keyword argument, but am willing to be convinced (@pytroll/core?). Would this code be easier if you converted the palette to a numpy array? I'm guessing maybe trollimage needs to be modified to take a list or numpy array as input to the colormap? Should the colormap class be updated to accept these modes? I think you could simplify/clean up the code (and feel better about it yourself) if you replaced the `_check_if_values` with a "what mode is this" and then in the for loop check if the mode starts with `"V"`. Just a thought.

Reply:
The palettes in the files are numpy arrays.  Existing code  used  np.load(filename) to read the files  so I just kept that.  I'm happy to use whatever keyword is desired .. "colormap_mode" seems reasonable.  I can modify re: checking for starting with "V" ... tks....

I'll leave to others the decision whether or not trollimage/colormap  should have utility routines to create colormaps from files/lists/numpy-arrays ... as opposed to routines within satpy which perform that function (at least for numpy arrays in a file) and provide trollimage with the colormap